test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -826,6 +826,68 @@ public void HubsCannotHaveOverloadedMethods()
             }
         }
 
+        [Theory]
+        [MemberData(nameof(HubTypes))]
+        public async Task SendToOthers(Type hubType)
+        {
+            var serviceProvider = CreateServiceProvider();
+
+            dynamic endPoint = serviceProvider.GetService(GetEndPointType(hubType));
+
+            using (var firstClient = new TestClient())
+            using (var secondClient = new TestClient())
+            {
+                Task firstEndPointTask = endPoint.OnConnectedAsync(firstClient.Connection);
+                Task secondEndPointTask = endPoint.OnConnectedAsync(secondClient.Connection);
+
+                await Task.WhenAll(firstClient.Connected, secondClient.Connected).OrTimeout();
+
+                await firstClient.SendInvocationAsync("SendToOthers", "To others").OrTimeout();
+
+                var secondClientResult = await secondClient.ReadAsync().OrTimeout();
+                var invocation = Assert.IsType<InvocationMessage>(secondClientResult);
+                Assert.Equal("Send", invocation.Target);
+                Assert.Equal("To others", invocation.Arguments[0]);
+
+                var firstClientResult = await firstClient.ReadAsync().OrTimeout();
+                var completionMessage = Assert.IsType<CompletionMessage>(firstClientResult);
+
+                // kill the connections
+                firstClient.Dispose();
+                secondClient.Dispose();
+
+                await Task.WhenAll(firstEndPointTask, secondEndPointTask).OrTimeout();
```

Comment:
My only minor concern is that a message could have been sent to `firstClient` and the test wouldn't fail, as far as I can tell. We should check that `.TryRead` returns false (or only returns Ping messages) **after** the endpoint tasks have concluded.

Reply:
I broadcast again after the others invocation and assert that that message is the first invocation message to the first client

Reply:
Ah, I see that now. Looks good then, thanks for clarifying.